cdap-ui/app/features/etlapps/controllers/create-ctrl.js

```diff
@@ -18,6 +17,29 @@ angular.module(PKG.name + '.feature.etlapps')
     $scope.etlSources = [];
     $scope.etlSinks = [];
     $scope.etlTransforms = [];
+    $scope.selectedEtlDraft = undefined;
+    $scope.etlDraftList = [];
+
+    $scope.onDraftChange = function(item, model) {
+      var filterFilter = $filter('filter'),
+          match = null,
+          swapObj = {};
+      if (!item) {
+        return; //un-necessary.
+      }
+      if ($scope.etlDrafts[item]) {
+        $scope.metadata = $scope.etlDrafts[item].config.metadata;
+        $scope.source = $scope.etlDrafts[item].config.source;
+        $scope.sink = $scope.etlDrafts[item].config.sink;
+        $scope.transforms = $scope.etlDrafts[item].config.transforms;
+      } else {
+        $scope.metadata.name = item;
+        $scope.metadata.type = $scope.metadata.type;
+        $scope.transforms = defaultTransforms;
+        $scope.source = defaultSource;
+        $scope.sink = defaultSink;
```

Comment:
We are starting to have a lot of $scope variables. Can we group them together and turn them into objects?